frontend/src/pages/AccountSend.vue

```diff
@@ -676,13 +688,31 @@ function useSendForm() {
     );
   }
 
+  function checkConfusables() {
+    const recipientIdString = recipientId.value || '';
+    try {
+      if (recipientIdString && recipientIdString.endsWith('eth')) {
+        assertNoConfusables(
```

Comment:
Yes exactly, you can definitely argue it is a premature abstraction. 

The argument for abstracting was all the things you mentioned and that this functionality felt more general as we can reuse this validation check for any input that may have confusables. And overriding the error message has definitely been useful as seen with all the other validation methods, but it could again be argued as premature.

I am fine to change it either way and will add tests as mentioned in another comment if kept.

Reply:
No, it isn't I will create an issue for the review page.

Reply:
#526